src/app/static/src/app/vue-chart/tests/bar/filterSelect.test.ts

```diff
@@ -126,8 +128,25 @@ describe("FilterSelect component", () => {
         vm.deselect({id: "fo2", label: "option 2"});
         await nextTick();
 
+        expect(wrapper.emitted("update:filter-select")!.length).toBe(1)
+        expect(wrapper.emitted("update:filter-select")![0][0]).toStrictEqual([]);
+
+        // In production the event above would commit a mutation which clears the value, simulating that here
+        // for the testing
+        await wrapper.setProps({value: []});
+
         await wrapper.setProps({isXAxis: false, isDisaggregateBy: false});
+        expect(wrapper.emitted("update:filter-select")!.length).toBe(2)
+        expect(wrapper.emitted("update:filter-select")![1][0]).toStrictEqual([{id: "fo1", label: "option 1"}]);
+    });
```

Comment:
Note the updates to this test, previously it would automatically select the first in the list of options, but now it will first reverts to the first of whatever is in the `value`. But if the value is empty this will then use the first from the list of options. In production there is a listener on the parent component which commits a mutation which updates the `value` whenever the `selection` is changed. So here the deselect would remove `id2` from the value. I've added another `setProps` which recreates this behaviour so we can still test that the first option is picked if the `value` prop is empty.